src/lib/run_environment.ml

```diff
@@ -94,10 +94,37 @@ module Tool = struct
   end
 end
 
+module Make_fun = struct
+  module Requirement = struct
+    type t = [
+      | `Processors of int
+      | `Internet_access
+      | `Memory of [ `GB of float | `Decent | `Big ]
+      | `Quick_run
+      | `Spark of string list (* custom parameters *)
+      | `Custom of string
+    ] [@@deriving yojson, show]
+  end
```

Comment:
IMO, the benefits of a regular variant are: simplicity (no more what's the 'chick' for questions) and concise type definitions that use these variants.